fall19/p10/test.py

```diff
@@ -618,24 +541,34 @@ def main():
     results = grade_answers(answer_cells)
     passing = sum(t['weight'] for t in results['tests'] if t['result'] == PASS)
     total = sum(t['weight'] for t in results['tests'])
-    results['score'] = 100.0 * passing / total
+
+    lint_msgs = lint(orig_notebook, verbose=1, show=False)
+    lint_msgs = filter(lambda msg: msg not in ALLOWED_LINT_ERRS, lint_msgs)
+    lint_msgs = list(lint_msgs)
+    results["lint"] = [str(l) for l in lint_msgs]
+
+    functionality_score = 90.0 * passing / total
+    linting_score = max(0.0, 10.0-len(lint_msgs))
```

Comment:
can we make linting score negative instead of weighting 0.9 against other tests?  Otherwise it's weird that they get 10% for turning in an empty notebook.

Reply:
Well, it also means they can have -10 then.
I'll min out the total score at 0 in that case.
I'm on it.

Reply:
Great, min'ing at 0 sounds like a good option, thanks!